### PR TITLE
Playercntl: Added tag based item restrictions

### PR DESCRIPTION
**Pull request by LazDisco**
Added tag based item restrictions. Functions the same way as items, but is dependent on the player's name. The idea behind this is too allow people to buy official faction ids without having to pester people to come online and give them an ID as all they need is the rename password.

### CODE REVIEW

**Comment by ConradWeiser** [comment]
Variable and logic structures seem to be variants of what already exists, nothing appears to be missing from the logic itself, looks great!

The only recommendation I have, is that you include an example of what the configuration changes look like somewhere in the Binaries/bin-conf folder.

Neat little addition.

**Comment by LazDisco** [comment]
Example done.

File: Binaries/bin-conf/flhook_plugins/playercntl.cfg [hunk 1]
```diff
@@ -73,6 +74,9 @@ shortname = playercntl
 maypause = false
 mayunload = true
 
+[TagItemRestrictions]
+dsy_license_admin = [Admin]; The admin id can now only be got by anyone with the [Admin] in their name
```
Comment (dsyalex) [inline Binaries/bin-conf/flhook_plugins/playercntl.cfg]: Tags are only restricted at the beginning of names

**Comment by LazDisco** [comment]
I've amended the code. Should now allow for multiple tags per item (split by comma), and I've tested it and can confirm that tags work no matter where in the name they are located.

**Comment by dsyalex** [comment]
I think you misunderstand - tags are only restricted *as prefixes*, so it is not secure to base access controls on tags simply being contained within character names - it must be a prefix match. Please see playercntl Rename::CreateNewCharacter

File: Plugins/Public/playercntl_plugin/PurchaseRestrictions.cpp [hunk 2]
```diff
@@ -236,6 +260,37 @@ namespace PurchaseRestrictions
 			}
 		}
 
+		// Restrict things via player tag
+		if (set_bCheckTagRestrictions)
+		{
+			for (map<uint, list<string>>::iterator iter = mapTagList.begin(); iter != mapTagList.end(); iter++)
+			{
+				if (iter->first == gbi.iGoodID)
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/PurchaseRestrictions.cpp]: direct lookup possible

File: Plugins/Public/playercntl_plugin/PurchaseRestrictions.cpp [hunk 3]
```diff
@@ -236,6 +260,28 @@ namespace PurchaseRestrictions
...
 		}
 
+		// Restrict things via player tag
+		if (set_bCheckTagRestrictions)
+		{
+			if (mapTagList.find(gbi.iGoodID) != mapTagList.end())
+			{
+				list<string> tags = mapTagList[gbi.iGoodID];
+				string scCharname = wstos((const wchar_t*)Players.GetActiveCharacterName(iClientID));
+				bool validTag = find(tags.begin(), tags.end(), scCharname) != tags.end();
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/PurchaseRestrictions.cpp]: This appears to make it such that the check is not for tags, the actual whole character name must be in the config?

**Comment by LazDisco** [comment]
I have deleted my FLHook fork and am now unable to make changes. Closing request.

**Pull request closed without merging**